lib/backend/buffer.go

```diff
@@ -84,10 +100,39 @@ func (c *CircularBuffer) Reset() {
 	}
 }
 
+func (c *CircularBuffer) SetInit() {
+	c.Lock()
+	defer c.Unlock()
+	if c.init {
+		return
+	}
+
+	var watchersToDelete []*BufferWatcher
+	c.watchers.walk(func(watcher *BufferWatcher) {
+		if ok := watcher.init(); !ok {
+			watchersToDelete = append(watchersToDelete, watcher)
+		}
+	})
+
+	for _, watcher := range watchersToDelete {
```

Comment:
I don't have too much context about how this is being used, but is it intended behavior to set `init` flag to true even if some watchers failed to initialize?

Reply:
Yes.  The purpose of the init state is to record the fact that the parent event stream (i.e. the event stream from the real backend) has been successfully established.  If an individual watcher can't receive its init event, that just means the individual watcher is unhealthy, and doesn't have any bearing on the overall init state of the buffer and its parent stream.